### PR TITLE
FS3373 Quickfix: Replace with triple-quoted interpolated string

### PR DESCRIPTION
**Pull request by seclerp**
This PR adds support for a quick fix on FS3373 error: 

image

To be aligned with https://github.com/dotnet/fsharp/pull/12945

### CODE REVIEW

File: ReSharper.FSharp/test/src/FSharp.Intentions.Tests/src/QuickFixes/ReplaceWithTripleQuotedInterpolatedStringFixTest.fs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+open NUnit.Framework
+
+type ReplaceWithTripleQuotedInterpolatedStringFixTest() =
+    inherit FSharpQuickFixTestBase<ReplaceWithTripleQuotedInterpolatedStringFix>()
+
+    override x.RelativeTestDataPath = "features/quickFixes/replaceWithTripleQuotedInterpolatedStringFix"
+
+    [<Test>] member x.``01 - Regular interpolated string``() = x.DoNamedTest()
+    [<Test>] member x.``02 - Verbatim interpolated string``() = x.DoNamedTest()
```
Comment (auduchinok) [inline ReSharper.FSharp/test/src/FSharp.Intentions.Tests/src/QuickFixes/ReplaceWithTripleQuotedInterpolatedStringFixTest.fs:12]: Could you please add tests for unfinished and multiline strings?

File: ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/ReplaceWithTripleQuotedInterpolatedStringFix.fs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+open JetBrains.ReSharper.Plugins.FSharp.Psi.PsiUtil
+
+type ReplaceWithTripleQuotedInterpolatedStringFix(error: SingleQuoteInSingleQuoteError) =
+    inherit FSharpQuickFixBase()
+
+    let getInterpolatedStringExpr (child: ITreeNode): IInterpolatedStringExpr =
+        let mutable node = getParent child
+        while node != null && not <| node :? IInterpolatedStringExpr do
+            node <- getParent node
+        node :?> _
```
[the remark this answers is not in the document]
Comment (seclerp) [reply ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/ReplaceWithTripleQuotedInterpolatedStringFix.fs]: Used `node.GetContainingNode<IInterpolatedStringExpr>()` instead

File: ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/ReplaceWithTripleQuotedInterpolatedStringFix.fs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+        let resultingNode: ITreeNode =
+            match tokenType with
+            | tokenType when tokenType == FSharpTokenType.VERBATIM_INTERPOLATED_STRING_START -> createStart content
+            | tokenType when tokenType == FSharpTokenType.VERBATIM_INTERPOLATED_STRING_MIDDLE -> createMiddle content
+            | tokenType when tokenType == FSharpTokenType.VERBATIM_INTERPOLATED_STRING_END -> createEnd content
+            | _ -> literal
+
+        if literal != resultingNode then
+            ModificationUtil.ReplaceChild(literal, resultingNode) |> ignore
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/ReplaceWithTripleQuotedInterpolatedStringFix.fs:66]: Could you remove the duplication here? The only meaningful difference is quotes replacement, which could be controlled by a param.

**Review by auduchinok — APPROVED** [review]
Thanks!

**Pull request merged**